### PR TITLE
hd: op: Configure Olmsted sensors following the hardware design

### PR DESCRIPTION
**Pull request by BonnieLo**
Summary:
- Identify card position and add sensor_number_interval to dynamic change sensor number. e.g. : SENSOR_NUM_2OU =  SENSOR_NUM_1OU + SENSOR_NUMBER_INTERVAL
- List change name table and use concatenation to dynamic change sensor name. e.g. : OP_2OU + _TEMP_C => OP_2OU_TEMP_C
- Refer to OPA sensor table "20221205" and OPB sensor table "20221201".

Test Plan:
- Check Sensor number and name : Pass

Test Log:
1. 1OU: uart:~$ platform sensor list_all ----------------------------------------------- [0x41] OP_1OU_E1S_SSD0_P12V_VOLT_V: ina233 [0x42] OP_1OU_E1S_SSD1_P12V_VOLT_V: ina233 [0x43] OP_1OU_E1S_SSD2_P12V_VOLT_V: ina233 ```
    [0x4d] OP_1OU_P1V8_ADC_VOLT_V   : adc
    [0x4e] OP_1OU_P0V9_ADC_VOLT_V   : adc
    [0x4f] OP_1OU_P1V2_ADC_VOLT_V   : adc

    -----------------------------------------------
2. 2OU: uart:~$ platform sensor list_all ----------------------------------------------- [0x71] OP_2OU_E1S_SSD0_P12V_VOLT_V: ina233 [0x72] OP_2OU_E1S_SSD1_P12V_VOLT_V: ina233 [0x73] OP_2OU_E1S_SSD2_P12V_VOLT_V: ina233 ``` [0x84] OP_2OU_E1S_SSD4_P12V_CURR_A: ina233 [0x89] OP_2OU_E1S_SSD3_P12V_PWR_W: ina233 [0x8a] OP_2OU_E1S_SSD4_P12V_PWR_W: ina233 -----------------------------------------------

3. 3OU:

    uart:~$ platform sensor list_all
    -----------------------------------------------
    [0xa1] OP_3OU_E1S_SSD0_P12V_VOLT_V: ina233
    [0xa2] OP_3OU_E1S_SSD1_P12V_VOLT_V: ina233
    [0xa3] OP_3OU_E1S_SSD2_P12V_VOLT_V: ina233
    ```
    [0xad] OP_3OU_P1V8_ADC_VOLT_V   : adc
    [0xae] OP_3OU_P0V9_ADC_VOLT_V   : adc
    [0xaf] OP_3OU_P1V2_ADC_VOLT_V   : adc
    -----------------------------------------------
4. 4OU:

    uart:~$ platform sensor list_all
    -----------------------------------------------
    [0xd1] OP_4OU_E1S_SSD0_P12V_VOLT_V: ina233
    [0xd2] OP_4OU_E1S_SSD1_P12V_VOLT_V: ina233
    [0xd3] OP_4OU_E1S_SSD2_P12V_VOLT_V: ina233
    ```
    [0xe4] OP_4OU_E1S_SSD4_P12V_CURR_A: ina233
    [0xe9] OP_4OU_E1S_SSD3_P12V_PWR_W: ina233
    [0xea] OP_4OU_E1S_SSD4_P12V_PWR_W: ina233
    -----------------------------------------------

### CODE REVIEW

File: meta-facebook/op2-op/src/platform/plat_sdr_table.c [hunk 1]
```diff
@@ -13,12 +13,1995 @@
...
+		0xD3, // LNRT
+		0xD5, // LCT
+		0xD7, // LNCT
+		0x00, // positive-going threshold
+		0x00, // negative-going threshold
+		0x00, // reserved
+		0x00, // reserved
+		0x00, // OEM
+		IPMI_SDR_STRING_TYPE_ASCII_8, // ID len, should be same as "size of struct"
+		"OP_1OU_P12V_EDGE_VOLT_V",
```
Comment (garnermic) [inline meta-facebook/op2-op/src/platform/plat_sdr_table.c:331]: Where is the P12V_EDGE voltage being read from?  This sensor is missing a location field.  Here is the text in our sensor naming specification to help with the name:

Definition: Should provide a sense on which device is being referenced or location within FRU
Example: Device + Power Rail or Location
Devices: CPU#, SOC#, ASIC#, DIMM#, PCH, NVME#, SSD#, HDD#, FAN#, HSC# (Hot Swap Controller), BMC (Baseboard Management Controller), BIC (Bridge IC), ADC (Analog-to-Digital Converter), VR (Voltage Regulator), CHS (Chassis), PMON (Power Monitoring IC)

**Review by garnermic — CHANGES_REQUESTED** [review]

**Comment by SaraLin-wiwynn** [comment]
Hi @garnermic 
We modify the sensor name according to your suggestion, thank you.
Would you have any considerations?

OPA (3OU is same)
OP_1OU_TEMP_C                       ->   OP_1OU_BIC_TEMP_C 
OP_1OU_RE_TIMER_TEMP_C              ->   OP_1OU_BIC_RE_TIMER_TEMP_C
OP_1OU_E1S_SSD0_TEMP_C 
OP_1OU_E1S_SSD1_TEMP_C
OP_1OU_E1S_SSD2_TEMP_C
OP_1OU_E1S_SSD0_P12V_VOLT_V 
OP_1OU_E1S_SSD1_P12V_VOLT_V 
OP_1OU_E1S_SSD2_P12V_VOLT_V
OP_1OU_P12V_EDGE_VOLT_V             ->   OP_1OU_BIC_P12V_EDGE_VOLT_V
OP_1OU_E1S_SSD0_P3V3_ADC_VOLT_V     ->   OP_1OU_E1S_ADC_SSD0_P3V3_VOLT_V
OP_1OU_E1S_SSD1_P3V3_ADC_VOLT_V     ->   OP_1OU_E1S_ADC_SSD1_P3V3_VOLT_V
OP_1OU_E1S_SSD2_P3V3_ADC_VOLT_V     ->   OP_1OU_E1S_ADC_SSD2_P3V3_VOLT_V
OP_1OU_P3V3_STBY_ADC_VOLT_V         ->   OP_1OU_ADC_P3V3_STBY_VOLT_V
OP_1OU_P1V8_ADC_VOLT_V              ->   OP_1OU_ADC_P1V8_VOLT_V                  
OP_1OU_P0V9_ADC_VOLT_V              ->   OP_1OU_ADC_P0V9_VOLT_V               
OP_1OU_P1V2_ADC_VOLT_V              ->   OP_1OU_ADC_P1V2_VOLT_V                     
OP_1OU_E1S_SSD0_P12V_CURR_A
OP_1OU_E1S_SSD1_P12V_CURR_A 
OP_1OU_E1S_SSD2_P12V_CURR_A 
OP_1OU_P12V_EDGE_CURR_A             ->   OP_1OU_E1S_P12V_EDGE_CURR_A
OP_1OU_E1S_SSD0_P12V_PWR_W
OP_1OU_E1S_SSD1_P12V_PWR_W
OP_1OU_E1S_SSD2_P12V_PWR_W
OP_1OU_P12V_EDGE_PWR_W              ->   OP_1OU_E1S_P12V_EDGE_PWR_W 

OPB (4OU is same)
OP_2OU_TEMP_C                       ->   OP_2OU_BIC_TEMP_C                                                                                     
OP_2OU_E1S_SSD0_TEMP_C                                                                             
OP_2OU_E1S_SSD1_TEMP_C     
OP_2OU_E1S_SSD2_TEMP_C
OP_2OU_E1S_SSD3_TEMP_C
OP_2OU_E1S_SSD4_TEMP_C 
OP_2OU_E1S_SSD0_P12V_VOLT_V
OP_2OU_E1S_SSD1_P12V_VOLT_V
OP_2OU_E1S_SSD2_P12V_VOLT_V
OP_2OU_E1S_SSD3_P12V_VOLT_V 
OP_2OU_E1S_SSD4_P12V_VOLT_V 
OP_2OU_MAIN_P12V_VOLT_V             ->   OP_2OU_BIC_MAIN_P12V_VOLT_V                                                    
OP_2OU_E1S_SSD0_P3V3_ADC_VOLT_V     ->   OP_2OU_E1S_ADC_SSD0_P3V3_VOLT_V                                                            
OP_2OU_E1S_SSD1_P3V3_ADC_VOLT_V     ->   OP_2OU_E1S_ADC_SSD1_P3V3_VOLT_V                                                            
OP_2OU_E1S_SSD2_P3V3_ADC_VOLT_V     ->   OP_2OU_E1S_ADC_SSD2_P3V3_VOLT_V                                                            
OP_2OU_E1S_SSD3_P3V3_ADC_VOLT_V     ->   OP_2OU_E1S_ADC_SSD3_P3V3_VOLT_V                                                            
OP_2OU_E1S_SSD4_P3V3_ADC_VOLT_V     ->   OP_2OU_E1S_ADC_SSD4_P3V3_VOLT_V                                                            
OP_2OU_P3V3_STBY_ADC_VOLT_V         ->   OP_2OU_ADC_P3V3_STBY_VOLT_V                                                        
OP_2OU_P1V8_STBY_ADC_VOLT_V         ->   OP_2OU_ADC_P1V8_STBY_VOLT_V                                                        
OP_2OU_P1V2_STBY_ADC_VOLT_V         ->   OP_2OU_ADC_P1V2_STBY_VOLT_V                                                        
OP_2OU_E1S_SSD0_P12V_CURR_A 
OP_2OU_E1S_SSD1_P12V_CURR_A
OP_2OU_E1S_SSD2_P12V_CURR_A
OP_2OU_E1S_SSD3_P12V_CURR_A
OP_2OU_E1S_SSD4_P12V_CURR_A
OP_2OU_MAIN_P12V_CURR_A             ->   OP_2OU_BIC_MAIN_P12V_CURR_A                                                    
OP_2OU_E1S_SSD0_P12V_PWR_W
OP_2OU_E1S_SSD1_P12V_PWR_W 
OP_2OU_E1S_SSD2_P12V_PWR_W
OP_2OU_E1S_SSD3_P12V_PWR_W
OP_2OU_E1S_SSD4_P12V_PWR_W
OP_2OU_MAIN_P12V_PWR_W              ->   OP_2OU_BIC_MAIN_P12V_PWR_W

**Comment by SaraLin-wiwynn** [comment]
Hi @garnermic 
Could you please confirm those sensor names?
Would you have any concerns?

**Comment by JeffShih-wiwynn** [comment]
Sensors' name are modified and this branch has been force updated.

**Comment by garnermic** [comment]
Changes merged as part of https://github.com/facebook/OpenBIC/pull/814.

**Pull request closed without merging**